File: app/src/main/java/com/wang/javamvp/base/rx/RxManager.java
package com.wang.javamvp.base.rx;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * @author Mis Wang
 * @date 2018/5/14  13:54
 * @fuction
 */
public class RxManager {
    private CompositeDisposable mCompositeDisposable = new CompositeDisposable();

    /**
     * 添加Disposable
     */
    public void add(Disposable disposable) {
        mCompositeDisposable.add(disposable);
    }

    /**
     * 解除绑定
     */
    public void dispose() {
        if (mCompositeDisposable != null) {
            mCompositeDisposable.dispose();
            mCompositeDisposable = null;
        }
    }
}

File: app/src/main/java/com/wang/kotlinmvp/base/rx/RxManager.kt
package com.wang.kotlinmvp.base.rx

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable


/**
 * @author Mis Wang
 * @date  2018/5/10  17:26
 * @fuction 用来统一管理Disposable
 */
class RxManager() {
    //声明Disposable容器
    private var mCompositeDisposable: CompositeDisposable? = CompositeDisposable()

    /**
     * 添加Disposable
     */
    fun add(disposable: Disposable) {
        mCompositeDisposable!!.add(disposable)
    }

    /**
     * 解除绑定
     */
    fun dispose() {
        mCompositeDisposable!!.dispose()
        mCompositeDisposable = null
    }
}